src/lib/plot/hint.js

```diff
@@ -23,11 +23,21 @@ import React from 'react';
 import PureRenderComponent from '../pure-render-component';
 import {getAttributeFunctor} from '../utils/scales-utils';
 
-const ORIENTATION_AUTO = 'auto';
-const ORIENTATION_TOPLEFT = 'topleft';
-const ORIENTATION_BOTTOMLEFT = 'bottomleft';
-const ORIENTATION_TOPRIGHT = 'topright';
-const ORIENTATION_BOTTOMRIGHT = 'bottomright';
+const ORIENTATION = {
+  AUTO: 'auto',
+  TOPLEFT: 'topleft',
+  BOTTOMLEFT: 'bottomleft',
+  TOPRIGHT: 'topright',
+  BOTTOMRIGHT: 'bottomright',
+  EDGETOP_LEFT: 'edgetopLeft',
```

Comment:
I just followed the style of naming that was part of this original API.  Orientation value follows this pattern:
  <top | bottom><left | right>

So, when Edgetop, Edgebottom, Edgeright, Edgeleft was introduced, I put an _ to allow better parsing of the constants name that respects the original naming scheme, that is:
  <edgetop | edgebottom | top | bottom>_<edgeleft | edgeright | left | right>

Since PR moving forwards introduce the constants, I did not change the original orientation string values by incorporating the _.  Apps using earlier version of this API choosing to update to this version would experience breaking change if they don't switch their use of the original string values 'topleft', 'topright', 'bottomleft', 'bottomright' to the constants.

I'll leave the naming as is because one might easily miss out the double underscore.

I'll add some comments to the code where the constants are defined that explains the constants.